src/component/reducer/preferences/actions/removeWorkspace.ts

```diff
@@ -16,12 +16,12 @@ export function removeWorkspace(
 
   let localData = getLocalStorage('nmr-general-settings');
   const storedWorkspaces = original(draft)?.workspaces || {};
-  const workspaces = Object.keys(storedWorkspaces).reduce((acc, key) => {
+  const workspaces = {};
+  Object.keys(storedWorkspaces).forEach((key) => {
     if (key !== workspace) {
-      acc[key] = storedWorkspaces[key];
+      workspaces[key] = storedWorkspaces[key];
     }
-    return acc;
-  }, {});
+  });
```

Comment:
You might be able to simply do `delete draft.workspaces[workspace]` here no?

Reply:
it shows an error `Do not delete dynamically computed property keys`

Reply:
I think the rule is valid if you are mutating the object, in which case you should use a `Map`. But since we are using immer, we are creating a new object, so the rule can be ignored.

In that case I think you can just ignore the rule with a `// eslint-disable-next-line ...` comment